kratos/utilities/intersection_utilities.h

```diff
@@ -237,6 +235,266 @@ class KRATOS_API(KRATOS_CORE) IntersectionUtilities
         return true;
     }
 
+    /**
+     * @brief Find the 3D intersection of a line (bounded) with a triangle (bounded) in the same plane
+     * @tparam TGeometryType The geometry type
+     * @tparam TCoordinatesType The type of coordinates
+     * @tparam TConsiderInsidePoints If considering inside points or just the intersections in the faces
+     * @param [in] rTriangleGeometry Is the tetrahedra to intersect
+     * @param [in] rLinePoint1 Coordinates of the first point of the intersecting line
+     * @param [in] rLinePoint2 Coordinates of the second point of the intersecting line
+     * @param [out] rIntersectionPoint1 The first intersection point coordinates
+     * @param [out] rIntersectionPoint2 The second intersection point coordinates
+     * @param [out] rSolution The intersection type index:
+     *         NO_INTERSECTION (disjoint - no intersection)
+     *         TWO_POINTS_INTERSECTION (intersect in two points)
+     *         ONE_POINT_INTERSECTION (intersect in one point)
+     * @param Epsilon The tolerance
+     */
+    template <class TGeometryType, class TCoordinatesType>
+    static void ComputeTriangleLineIntersectionInTheSamePlane(
+        const TGeometryType& rTriangleGeometry,
+        const TCoordinatesType& rLinePoint1,
+        const TCoordinatesType& rLinePoint2,
+        TCoordinatesType& rIntersectionPoint1,
+        TCoordinatesType& rIntersectionPoint2,
+        int& rSolution,//IntersectionUtilitiesTetrahedraLineIntersectionStatus& rSolution,
```

Comment:
```suggestion
        int& rSolution,//IntersectionUtilitiesTetrahedraLineIntersectionStatus& rSolution,
```
Wouldn't it be better to return the intersection status rather than passing it as a reference? This is what we normally do in the other methods.

Reply:
This method is used as an auxiliary method as well, must be as a reference

Reply:
But why needs to be a reference when is used as an auxiliary method?

Reply:
Because the previous information is relevant to determine the intersection status

Reply:
Now also returns the int

Reply:
OK. Now I understand your point. Thanks.